frontend/packages/topology/src/components/quick-search/QuickSearchList.tsx

```diff
@@ -0,0 +1,117 @@
+  totalItems: number;
+  selectedItemId: string;
+  searchTerm: string;
+  namespace: string;
+  onSelectListItem: (itemId: string) => void;
+}
+
+const QuickSearchList: React.FC<QuickSearchListProps> = ({
+  listItems,
+  totalItems,
+  selectedItemId,
+  searchTerm,
+  namespace,
+  onSelectListItem,
+}) => {
+  const { t } = useTranslation();
+  const ref = React.useRef(null);
+
+  const openForm = (e: React.SyntheticEvent) => {
+    e.persist();
+    e.stopPropagation();
+    e.preventDefault();
+    const targetItemUID = ref?.current?.props?.id;
+    const item = listItems.find((listItem) => listItem.uid === targetItemUID);
+    removeQueryArgument('catalogSearch');
+    history.push(item?.cta?.href);
+  };
+
+  const goToCatalogPage = () => {
+    removeQueryArgument('catalogSearch');
+    history.push(`/catalog/ns/${namespace}?keyword=${searchTerm}`);
+  };
+
+  return (
+    <DataList
+      className="odc-quick-search-list"
+      aria-label="Catalog Items"
+      selectedDataListItemId={selectedItemId}
+      onSelectDataListItem={(itemId) => itemId !== 'viewAll' && onSelectListItem(itemId)}
+      isCompact
+    >
+      {listItems.map((item) => (
+        <DataListItem
+          id={item.uid}
+          key={item.uid}
+          className={cx('odc-quick-search-list__item', {
+            'odc-quick-search-list__item--highlight': item.uid === selectedItemId,
+          })}
+          onDoubleClick={openForm}
+          ref={ref}
+        >
+          <DataListItemRow className="odc-quick-search-list__item-row">
+            <DataListItemCells
+              className="odc-quick-search-list__item-content"
+              dataListCells={[
+                <DataListCell isIcon>
+                  <img className="odc-quick-search-list__item-icon" src={item.icon?.url} alt="" />
+                </DataListCell>,
+                <DataListCell style={{ paddingTop: '6px' }} width={2} wrapModifier="truncate">
+                  <span className="odc-quick-search-list__item-name">{item.name}</span>
+                  <Split style={{ alignItems: 'center' }} hasGutter>
+                    <SplitItem>
+                      <Label>{item.type}</Label>
+                    </SplitItem>
+                    <SplitItem>
+                      <TextContent>
+                        <Text component={TextVariants.small}>{item.provider}</Text>
+                      </TextContent>
+                    </SplitItem>
+                  </Split>
+                </DataListCell>,
+              ]}
+            />
+          </DataListItemRow>
+        </DataListItem>
+      ))}
+      <DataListItem id="viewAll" onClick={goToCatalogPage}>
+        <DataListItemRow className="odc-quick-search-list__all-items-link">
+          <DataListItemCells
+            dataListCells={[
+              <DataListCell>
+                <Button variant="link">
+                  {t('topology~View all results for')} &quot;{searchTerm}&quot; ({totalItems})
```

Comment:
I think translations should have the swap terms in them 🤔 

It's been a long time since I dealt with I18n, but I think some languages are structured in such a fashion that the full sentence structure could flip around on you and you'd lose order of variables.

Reply:
I am not sure if I understand this completely but you mean swap terms for the variables or the entire sentence?

Reply:
@divyanshiGupta A crude example:

In English: "I will be 25 on Thursday 1234"
In German: "Ich werde am Donnerstag 1234 25 sein"

It does not always pan out that your numbers and context of the phrase are always going to be at the end of a phrase in other languages. So we need to inject the variables into the phrase in order to allow translators to re-arrange to meet proper grammatical structure for the given language.

Reply:
Got it. Thanks!